ietf/iesg/views.py

```diff
@@ -528,4 +531,48 @@ def photos(request):
         role.last_initial = role.person.last_name()[0]
     return render(request, 'iesg/photos.html', {'group_type': 'IESG', 'role': '', 'roles': roles })
 
-    
\ No newline at end of file
+def month_choices():
+    choices = [(str(n).zfill(2), str(n).zfill(2)) for n in range(1, 13)]
+    choices.insert(0, ('', 'Select Month...'))
+    return choices
+
+def year_choices():
+    this_year = datetime.date.today().year
+    choices = [(str(n), str(n)) for n in range(this_year, 2009, -1)]
+    choices.insert(0, ('', 'Select Year...'))
+    return choices
+
+class ActivityForm(forms.Form):
+    month = forms.ChoiceField(choices=month_choices, required=False)
+    year = forms.ChoiceField(choices=year_choices, required=False)
+
+    def clean(self):
+        super().clean()
+        month = self.cleaned_data['month']
+        year = self.cleaned_data['year']
+        if not (month.isdigit() and year.isdigit()):
+            raise ValidationError('Must select a month and a year')
+
+        return self.cleaned_data
+
+def ietf_activity(request):
+    # default date range for last month
+    today = datetime.date.today()
+    first_day_this_month = today + relativedelta.relativedelta(day=1)
+    edate = first_day_this_month - relativedelta.relativedelta(days=1)
+    sdate = edate + relativedelta.relativedelta(day=1)
+    if request.method == 'GET':
+        form = ActivityForm(request.GET)
+        if form.is_valid():
+            month = form.cleaned_data['month']
+            year = form.cleaned_data['year']
+            sdate = datetime.date(int(year), int(month), 1)
+            next_month = sdate + relativedelta.relativedelta(months=1)
+            edate = next_month - relativedelta.relativedelta(days=1)
```

Comment:
I think you have a fencepost error on your end date here and in the earlier default case. The `get_activity_stats()` method selects items with `time__gte=<midnight on sdate>` and `time__lt=<midnight on edate>`, so I think you'd want to pass 2023-02-01 and 2023-03-01 to get all of Feb 2023.